### ISSUE TITLE
Release v0.8.0

### ISSUE THREAD

**Issue by marquiz** [kubernetes-sigs/node-feature-discovery#437]
## Release Checklist

- [x] All [OWNERS](https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/OWNERS) must LGTM the release proposal
- [x] Verify that the changelog in this issue is up-to-date
- [x] For major releases (v0.$MAJ.0), an OWNER creates a release branch with
      `git branch release-0.$MAJ master`
- [x] Prepare `release-0.$MAJ` release branch
  - [x] An OWNER creates a vanilla release branch from master and pushes it with
        `git push release-0.$MAJ`
  - [x] Run `scripts/prepare-release.sh $VERSION` to turn references to point to the upcoming release
        (README, deployment templates, docs configuration, test/e2e flags), submit a PR agains the release branch
  - An OWNER prepares a draft release
    - [x] Create a draft release at [Github releases page](https://github.com/kubernetes-sigs/node-feature-discovery/releases)
    - [x] Write the change log into the draft release
    - [x] Upload release artefacts generated by `prepare-release.sh` script above to the draft release
  - [x] An OWNER runs
       `git tag -s $VERSION`
        and inserts the changelog into the tag description.
- [x] An OWNER pushes the tag with
      `git push $VERSION`
  - Triggers prow to build and publish a staging container image
      `gcr.io/k8s-staging-nfd/node-feature-discovery:$VERSION`
  - Triggers build of the documentation and publish it at
        https://kubernetes-sigs.github.io/node-feature
- [x] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io), updating `k8s.gcr.io/images/k8s-staging-nfd/images.yaml` to promote the container image to production
- [x] Wait for the PR to be merged and verify that the image (`k8s.gcr.io/nfd/node-feature-discovery:$VERSION`) is available.
- [x] Publish the draft release prepared at the [Github releases page](https://github.com/kubernetes-sigs/node-feature-discovery/releases)
- [x] Add a link to the tagged release in this issue.
- [x] Send an announcement email to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] node-feature-discovery $VERSION is released`
- [x] Add a link to the release announcement in this issue
- [x] For a major release, update README in master branch
  - [x] Update references to point to `$VERSION` and submit a PR
  - [x] Wait for the PR to be merged, this will also update the Helm
        repo index to be updated to contain the latest release
- [x] Close this issue

#### NFD Helm repository

NFD now hosts a Helm chart repository as a new deployment option. Installation of the latest release can be as simple as:

```
helm repo add nfd http://kubernetes-sigs.github.io/node-feature-discovery/charts
helm repo update
helm install nfd/node-feature-discovery --namespace node-feature-discovery --create-namespace --generate-name
```

See the [documentation](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/get-started/deployment-and-usage.html#deployment-with-helm) for more detailed instructions.

#### Dynamic TLS certificate updates

Support for dynamic TLS certificate updates was added. This enables e.g. the usage of [cert-manager](https://cert-manager.io/docs/) for certificate rotation. See the [documentation](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/get-started/deployment-and-usage.html#automated-tls-certificate-management-using-cert-manager) for more details.

#### Improved worker configuration

Now most of the nfd-worker options (e.g. sleep-interval) are dynamically configurable via the configuration file. Most of the corresponding command line flags have been marked as deprecated. See [documentation of the core config options](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/advanced/worker-configuration-reference.html#core) for more details.

#### Minimal container image variant

NFD now has a "minimal" variant of container image, based on distroless/base. The minimal image provides enhanced security and reduced image footprint, but, limits the nfd-worker hook support to statically linked binaries only. See the [documentation](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/get-started/deployment-and-usage.html#minimal) for more details.

#### Support multiple parallel NFD instances

New `-instance` command line flag for nfd-master was added, making it possible to run multiple deployments of NFD in parallel.

#### Nodename rule to custom source

A new `nodename` rule was added to the quiver to enable creation of user-specific labels based on node name. See the [documentation](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/get-started/features.html#nodename-rule) for detailed description and examples.

#### Additional CPU pstate and cstate features

New CPU features labels were added:
- feature.node.kubernetes.io/cpu-pstate.status
- feature.node.kubernetes.io/cpu-pstate.scaling_governor
- feature.node.kubernetes.io/cpu-cstate.enabled

#### Klog for logging

NFD switched to klog-based logging, greatly improving logging and debugging capabilities. Moreover, most of the logger related options in nfd-worker are dynamically adjustable through the config file, making it possible to e.g. control the verbosity of nfd-worker instances at run-time.

### Miscellaneous

- Switch to flags package in command line arg parsing (#444)
- Update `klauspost/cpuid` to v2.0.2 (#421)
- nfd-worker: stop masking crashes in feature discovery (#458)

### List of PRs

- Cherry-pick: github: update new-release issue template (#416)
- github: workflow fixes (#419)
- README: update references to v0.7.0 (#420)
- go.mod: update to klauspost/cpuid/v2@v2.0.2 (#421)
- Set imagePullPolicy in the deployment templates to Always (#430)
- Change the worker conf update mechanism of templates (#434)
- scripts/test-infra: update golangci-lint to v1.36.0 (#436)
- Replace improper usage of filepath.Join with path.Join (#435)
- Update gogo/protobuf and golang.org/x/text (#438)
- nfd-worker: use fsnotify for watching for config file changes (#432)
- nfd-master: implement --instance flag (#431)
- Update OWNERS file (#441)
- nfd-worker: add core config (#433)
- Add helm chart for NFD (#423)
- Add NFD master '--instance' flag support to Helm chart (#443)
- Fix daemonset deployment template (#446)
- Fix NFD master chart template (#447)
- Switch to flags package in command line arg parsing (#444)
- Added nodename rule to custom source (#429)
- docs: align docs with the single-dash command line flags (#449)
- docs: enable markdown linting (#451)
- nfd-master: switch to klog (#445)
- nfd-worker: switch to klog (#450)
- test/e2e: make openshift bits optional (#452)
- Logging fixes (#453)
- scripts/test-infra: enable helm linting (#455)
- nfd-worker: stop masking crashes in feature discovery (#458)
- logs should all be lower case when possible (#454)
- source: drop stale BoolFeatureValue type (#459)
- Makefile: increase golangci-lint timeout to 7min (#461)
- nfd-worker: improve log messages of config handling (#460)
- pkg/utils: fix possible segfault in RegexpVal.Set (#462)
- github: limit gh-pages build of tags to (final) semver releases (#466)
- logging: set grpc to use klog for logging (#465)
- Dockerfile: add minimal image (#469)
- docs: unify capitalization of headings (#470)
- Support TLS certificate updates (#442)
- pkg/utils: show correct source file in gRPC logs (#475)
- Support for additional cpu features (#463)
- use cert-manager for TLS (#379)
- docs: describe deployment of the minimal image (#471)
- nfd-worker: don't log labels returned by sources by default (#474)
- docs: updates to documentation of Helm parameters (#472)
- docs: fix some internal references (#480)
- Host NFD Helm repo in gh-pages (#457)
- github: update new-release issue template assignees (#482)
- docs: document all config file options (#481)
- Remove --sleep-interval flag from worker deployment templates (#483)
- source/custom: dump config in more human-readable form (#473)

## Prerequisites/Roadmap

With these in place we would have enough content to justify a new release. 

- [x] `--instance` flag (#431)
- [x] nfd-worker config fsnotify (#432)
- Helm charts
  - [x] add Helm charts to NFD (#423)
  - [x] support '--instance' flag (#443)
  - [x] Helm repo (#457)
  - [x] documentation updates (#472)
- [x] hostname-based labeling (#429)
- [x] nfd-worker core config (#433)
- [x] additional cpu features (#439)
- [x] cert-manager support (#379)
- [x] document *minimal* image (#471)

**Comment by zvonkok** [comment]
@marquiz Looking good so far, but I am missing one PR that should have been posted. Do we have some time to include another PR that is "crucial" for the next release that will be used to downstream it into OpenShift?

**Comment by marquiz** [comment]
> @marquiz Looking good so far, but I am missing one PR that should have been posted. Do we have some time to include another PR that is "crucial" for the next release that will be used to downstream it into OpenShift?

Yeah, sure. That list above was just an initial proposal for a minimum that would be nice to get. Would it be possible to create an issue about that and add it to the list above?

**Comment by zvonkok** [comment]
@marquiz Please add #439 to the release listing, thanks.

**Comment by marquiz** [comment]
> @marquiz Please add #439 to the release listing, thanks.

Done

**Comment by ArangoGutierrez** [comment]
#433 merged ✅

**Comment by ArangoGutierrez** [comment]
https://github.com/kubernetes-sigs/node-feature-discovery/pull/423 merged ✅

**Comment by e0ne** [comment]
Can we include #443 into the release too? It's small and useful patch to finish #431 and #423 integration

**Comment by marquiz** [comment]
> Can we include #443 into the release too? It's small and useful patch to finish #431 and #423 integration

Yes, of course. Makes definitely sense. Added to the list

**Comment by e0ne** [comment]
Thank you, @marquiz!

**Comment by marquiz** [comment]
Added few more open PRs to the "roadmap": Heml repo plus the docs updates

**Comment by ArangoGutierrez** [comment]
We are 2 PR's away from Release! 
-  Helm repo (#457)
-  Documentation updates (#472)

**Comment by marquiz** [comment]
> We are 2 PR's away from Release!

Yeah 😄  I updated the changelog, making the *Major* section quite a bit more verbose, briefly explaining each major change

**Comment by marquiz** [comment]
Only #457 to go

**Comment by marquiz** [comment]
Heh, one more latecomer 🙄  It'd be nice to get #481. It's just documentation

**Comment by ArangoGutierrez** [comment]
well 😃 
#481 is merged!

**Comment by ArangoGutierrez** [comment]
Sounds like release week to me

**Comment by zvonkok** [comment]
LGTM

**Comment by zvonkok** [comment]
What a great release, kudos to all involved and especially to @marquiz for letting us work on such a great piece of software! :)

**Comment by Ethyling** [comment]
LGTM

**Comment by adrianchiris** [comment]
LGTM

**Comment by marquiz** [comment]
Release page:
https://github.com/kubernetes-sigs/node-feature-discovery/releases/tag/v0.8.0

Release announcement
https://groups.google.com/g/kubernetes-dev/c/Ig2vCTmtUhE

**Comment by marquiz** [comment]
Big thanks to everybody involved, we finally have a release!
